deployments/pulse3d/services/pulse3d/src/main.py

```diff
@@ -311,3 +318,64 @@ async def soft_delete_jobs(
     except Exception as e:
         logger.error(f"Failed to soft delete jobs: {repr(e)}")
         raise HTTPException(status_code=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+@app.post("/jobs/download")
+async def download_analyses(
+    request: Request,
+    details: DownloadRequest,
+    token=Depends(ProtectedAny(scope=["users:free"])),
+):
+
+    jobs = details.jobs
+    user_id = str(uuid.UUID(token["userid"]))
+    customer_id = str(uuid.UUID(token["customer_id"]))
+
+    # check if for some reason an empty list was sent
+    if not jobs:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="Nothing to download.",
+        )
+
+    try:
+        unique_filenames = list()
+        keys = list()
+
+        for job in jobs:
+            upload_id = job.uploadId
+            job_id = job.jobId
+            filename = job.analyzedFile
+            keys.append(f"analyzed/{customer_id}/{user_id}/{upload_id}/{job_id}/{filename}")
+
+            if filename in unique_filenames:
+                # grabs index of file in list of duplicate filenames to append to filename to differentiate
+                duplicate_filename = [j.jobId for j in jobs if j.analyzedFile == filename]
+                idx = duplicate_filename.index(job_id)
+                # add duplicate index to differentiate duplicate filenames
+                root, ext = os.path.splitext(filename)
+                filename = "".join([f"{root}_({idx})", ext])
+
+            unique_filenames.append(filename)
+
+        # Grab ZIP file from in-memory, make response with correct MIME-type
+        return StreamingResponse(
+            content=stream_zip(
+                _yield_s3_objects(bucket=PULSE3D_UPLOADS_BUCKET, keys=keys, filenames=unique_filenames)
+            ),
+            media_type="application/zip",
+        )
+
+    except Exception as e:
+        logger.error(f"Failed to download analyses: {repr(e)}")
+        raise HTTPException(status_code=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+
+def _yield_s3_objects(bucket: str, keys: List[str], filenames: List[str]):
+    try:
+        s3 = boto3.session.Session().resource("s3")
+        for (idx, key) in enumerate(keys):
+            obj = s3.Object(bucket_name="test-pulse3d-uploads", key=key)
```

Comment:
you'll want to grab the bucket name from the env so this will work on test and prod

Reply:
Good catch, just checking  in test if the FE streamsaver package will work with statically generated pages

Reply:
time has dropped from 1.5 minutes to .4 min

Reply:
I get this build error: 
`Collecting page data ...streamsaver is meant to run on browsers main thread`
```
> Build error occurred
ReferenceError: document is not defined
    at /Users/lucipak/Documents/work/CuriBio/k8s_curi/frontend/pulse3d/src/node_modules/streamsaver/StreamSaver.js:24:66
    at /Users/lucipak/Documents/work/CuriBio/k8s_curi/frontend/pulse3d/src/node_modules/streamsaver/StreamSaver.js:7:24
    at Object.<anonymous> (/Users/lucipak/Documents/work/CuriBio/k8s_curi/frontend/pulse3d/src/node_modules/streamsaver/StreamSaver.js:11:3)
```

might try adding to the service worker?